### PR TITLE
Omar/delete test by id

### PR DESCRIPTION
**Pull request by obaranek**
## Notion ticket link

[Ticket Name](https://www.notion.so/uwblueprintexecs/Delete-test-0cce32099ffb4bc99409082ebd983a00)

## Implementation description
* created interface for deleteTest
* implemented deleteTest
* added two tests to test deleteTest behaviour when test id is found and is missing.

## Steps to test
1.run yarn test and make sure both tests, "deleteTest" and "deleteTest" when id not found", pass

## What should reviewers focus on?
* 

## Checklist
- [ ] My PR name is descriptive and in imperative tense
- [ ] My commit messages are descriptive and in imperative tense. My commits are atomic and trivial commits are squashed or fixup'd into non-trivial commits
- [ ] I have run the appropriate linter(s)
- [ ] I have requested a review from the PL, as well as other devs who have background knowledge on this PR or who will be building on top of this PR

### CODE REVIEW

File: backend/services/interfaces/testService.ts [hunk 1]
```diff
@@ -29,4 +29,12 @@ export interface ITestService {
    * @throws Error if creation fails
    */
   createTest(test: CreateTestRequestDTO): Promise<TestResponseDTO>;
+
+  /**
+   * delete a TestSession with the given id, return deleted id
```
Comment (carissa-tang) [inline backend/services/interfaces/testService.ts]:
```suggestion
   * delete a Test with the given id, return deleted id
```

**Review by carissa-tang — COMMENTED** [review]
are we doing soft deletes?

**Comment by obaranek** [comment]
> are we doing soft deletes?

@joyce-shi

**Comment by joyce-shi** [comment]
> > are we doing soft deletes?
> 
> @joyce-shi

let's do hard deletes for the sake of simplicity right now. if we find it's an issue we can revisit in the future

File: backend/services/implementations/testService.ts [hunk 2]
```diff
@@ -40,6 +40,21 @@ class TestService implements ITestService {
...
+  async deleteTest(id: string): Promise<string> {
+    try {
+      const deletedTest = await MgTest.findByIdAndDelete(id);
+      if (!deletedTest) {
+        throw new Error(`Test ${id} not found`);
+      }
+      return id;
+    } catch (error: unknown) {
+      Logger.error(
+        `Failed to delete entity. Reason = ${getErrorMessage(error)}`,
```
Comment (joyce-shi) [inline backend/services/implementations/testService.ts]: nit: let's change this to `Failed to delete test` for clarity

**Review by joyce-shi — CHANGES_REQUESTED** [review]
LGTM, left a small comment re: error clarity.
One quick thing: I noticed that in this PR, there's a lot of changes highlighted, where there aren't really any changes (i.e spacing, line breaks). Just wanted to double check that you ran the linter before opening this?

**Review by joyce-shi — APPROVED** [review]

**Pull request merged**